dotnet-7.yaml

```diff
@@ -94,6 +94,8 @@ subpackages:
     dependencies:
       runtime:
         - dotnet-7
+      provides:
+        - dotnet-runtime=7
```

Comment:
This will technically prevent both branches of dotnet from being installed at the same time, even though Microsoft designed it so you can.  Are we sure we want that?

Reply:
I can split out a shim subpackage for this.

Reply:
ok, these are distinct subpackages now